hub/instruction_handling/storage/constraints.tex

```diff
@@ -221,31 +230,87 @@
 				\]
 		\end{description}
 	\item[\underline{Setting the refund:}]
-		recall that among instructions raising the $\stackDecStoFlag$ only the \inst{SSTORE} instruction may grand refunds.
-		Futhermore refunds may only be granted in non reverting contexts.
+		recall that among instructions raising the $\stackDecStoFlag$ only the \inst{SSTORE} instruction ($\locIsSstore \equiv 1$) may grand refunds.
+		Futhermore refunds may only be granted in non reverting contexts ($\cnWillRev \equiv 0$.)
 		See section~(\ref{hub: generalities: refunds: refunds}) for more details.
 
-		\If \Big($\cnWillRev_{i} = 0$ \et $\locIsSstore = 1$\Big) \Then we thus impose that
-		\begin{enumerate}
-			\item \If $\locNextIsCurr = 1$ \Then $\refund\new_{i} = \refund_{i}$
-			\item \If $\locNextIsCurr = 0$ \Then
-				\begin{enumerate}
-					\item \If $\locCurrIsOrig = 0$ \Then
-						\[
-							\refund\new_{i} = \refund_{i} 
-							+
-							\left[ \begin{array}{crcccl}
-								+ & (1 - \locOrigIsZero) & \cdot & (\locNextIsZero - \locCurrIsZero) & \cdot & R_\text{sclear}                                 \\
-								+ & \locNextIsOrig       & \cdot & \locOrigIsZero                    & \cdot & \big(G_\text{sset} - G_\text{warmaccess}\big)   \\
-								+ & \locNextIsOrig       & \cdot & (1 -\locOrigIsZero)               & \cdot & \big(G_\text{sreset} - G_\text{warmaccess}\big) \\
-							\end{array} \right]
-						\]
-						\item \If $\locCurrIsOrig = 1$ \Then
-							\begin{enumerate}
-								\item \If $\locNextIsZero = 0$ \Then $\refund\new_{i} = \refund_{i}$
-								\item \If $\locNextIsZero = 1$ \Then $\refund\new_{i} = \refund_{i} + R_\text{sclear}$
-							\end{enumerate}
-				\end{enumerate}
-		\end{enumerate}
-\end{description}
-
+		We therefore unconditionally impose
+		\[
+			\refund\new _{i} = \refund_{i} + (1 - \cnWillRev_{i}) \cdot \locIsSstore \cdot
+			\left[ \begin{array}{cr}
+				+ & \locCleanClear \\
+				+ & \locDirtyClear \\
+				+ & \locDirtyReset \\
+			\end{array} \right]
+		\]
+		where $\locCleanClear$, $\locDirtyClear$, $\locDirtyReset$ are shorthands which we define below.
+		
+		\saNote{}
+		In the explanations provided below we use notations from the \cite{EYP} whereby
+		$v_0$ stands for the value originally in storage,
+		$v$ for the value currently in storage,
+		$v'$ for the value we are attempting to store:
+		\begin{description}
+			\item[\underline{Defining \locCleanClear{}:}]
+				we set
+				\[
+					\locCleanClear \define
+					\left[ \begin{array}{cr}
+						\cdot & \locNextNotCurr \\
+						\cdot & \locCurrIsOrig       \\
+						\cdot & \locNextIsZero       \\
+					\end{array} \right]
+					\cdot R_\text{sclear}
+				\]
+				which corresponds to
+				\begin{itemize}
+					\item $R_\text{sclear}$ whenever $[v \neq v'] \wedge [v = v_{0}] \wedge [v' = 0]$
+					\item and $0$ otherwise
+				\end{itemize}
+			\item[\underline{Defining \locDirtyClear{}:}]
```

Comment:
There is an issue with the +ve case for R_sclear in the r_dirtyclear ---  I think you are missing a case.  You have said that:

* +Rsclear whenever `[v != v']` and `[v0 != v]` and `[v0 != 0]` and `[v' = 0]`

But, you are only enforcing three of these cases.  For the -Rsclear case, that seems fine as one of them is implied.  However, for the +Rsclear case this does not follow:

*  `[v0 != v]` and `[v0 != 0]` and `[v' = 0]` implies `[v != v']`

I didn't look at the r_dirtyreset case, but it may be similar.

Reply:
>But, you are only enforcing three of these cases.

I believe I'm enforcing all four of them. The first term is a **product** of two terms, then there is the middle term, and then there is either `+1` if `[v == 0]` or `-1` if `[v' == 0]`. So there are 3 common conditions and then one of two possible final conditions.

![image](https://github.com/user-attachments/assets/e5b02c32-46d1-482a-8410-0f40147a703e)

Reply:
I can re-typeset it if you find it unclear.

Reply:
Hmmmm, let me think about this.  It looks like this:

```
(v' != v) * (v != v0) * (v0 != 0) * (-v=0 + v'=0) * Rsclear
```
Ok, let's break up the `+` into the two cases:

* **[-ve]** (v' != v) * (v != v0) * (v0 != 0) * (v=0) * -Rsclear
* **[+ve]** (v' != v) * (v != v0) * (v0 != 0) * (v'=0) * +Rsclear

Ok, I see.  Maybe that does look better now.  I was missing that the first one was a product.  

You could re-typeset it ... but I guess that's upto you.